File: week3/3.6/ObjC/ObjC/ViewController.m
//
//  ViewController.m
//  ObjC
//
//  Created by Eric Lanza on 7/21/18.
//  Copyright © 2018 ETLanza. All rights reserved.
//

#import "ViewController.h"
#import "DMNPerson.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    DMNPerson *person = [[DMNPerson alloc] initWithName:@"Steve" age:62 height:6];
    person.height = 5.5;
    
    [person sayYourName];
    DMNPerson *friend = [[DMNPerson alloc] initWithName:@"Jony" age:50 height:6.2];
    [person driveCar:@"Yellow Corvet" to:@"New York" bringingAlong:friend];
}

@end

File: week2/2.3/CoreDataChallenge/CoreDataChallenge/ViewController.swift
//
//  ViewController.swift
//  CoreDataChallenge
//
//  Created by Eric Lanza on 7/10/18.
//  Copyright © 2018 ETLanza. All rights reserved.
//

import UIKit
import CoreData

class ViewController: UIViewController {
    
    
    @IBOutlet weak var personNameTextField: UITextField!
    
    @IBAction func saveButtonTapped(_ sender: Any) {
        guard let name = personNameTextField.text else { return }
        _ = Person(name: name)
        saveToPersistentStore()
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        var person: [Person] {
            let request: NSFetchRequest<Person> = Person.fetchRequest()
            do {
                return try CoreDataStack.context.fetch(request)
            } catch let error {
                print("Error fetching person: \(error)")
                return []
            }
        }
 
        personNameTextField.text = person.first?.name
    }
    
    private func saveToPersistentStore() {
        do {
            try CoreDataStack.context.save()
        } catch let error {
            print("Error saving to persistent store: \(error.localizedDescription)")
        }
    }
    
}


